src/CSharp/CodeCracker/Design/InconsistentAccessibility/InconsistentAccessibilityCodeFixProvider.cs

```diff
@@ -0,0 +1,159 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CodeCracker.CSharp.Design.InconsistentAccessibility
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(InconsistentAccessibilityCodeFixProvider)), Shared]
+    public sealed class InconsistentAccessibilityCodeFixProvider : CodeFixProvider
+    {
+        internal const string InconsistentAccessibilityInMethodParameterCompilerErrorNumber = "CS0051";
+
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(InconsistentAccessibilityInMethodParameterCompilerErrorNumber);
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics.First();
+
+            var inconsistentAccessibilityProvider = GetInconsistentAccessibilityProvider(diagnostic);
+
+            var info = await inconsistentAccessibilityProvider.GetInconsistentAccessibilityInfoAsync(context.Document, diagnostic, context.CancellationToken).ConfigureAwait(false);
+
+            if(info.TypeToChangeAccessibility != null)
+            {
+                var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+                var typeSymbol = semanticModel.GetSymbolInfo(info.TypeToChangeAccessibility, context.CancellationToken).Symbol;
+
+                if(typeSymbol != null)
+                {
+                    var document = context.Document.Project.Solution.GetDocument(typeSymbol.Locations[0].SourceTree);
+
+                    if(document != null)
+                    {
+                        context.RegisterCodeFix(CodeAction.Create(info.CodeActionMessage, c => ChangeTypeAccessibilityInDocument(document, info.NewAccessibilityModifiers, typeSymbol.Locations[0].SourceSpan, c)), diagnostic);
+                    }
+                }
+            }
```

Comment:
`RegisterCodeFixesAsync` checks whether type declaration exists in solution.
I think we can't move this to `CodeAction.Create` because if the type declaration is only in the metadata we shouldn't propose code fix at all.

Actually we can move part that finds Document and in `RegisterCodeFixesAsync` only see wheter any `Location` is in source. Agree?